src/dsp/pdr.c

```diff
@@ -1655,3 +1737,501 @@ int pldm_entity_association_pdr_create_new(pldm_pdr *repo,
+{
+	assert(repo != NULL);
+
+	uint32_t delete_handle = 0;
+	pldm_pdr_record *record = repo->first;
+	pldm_pdr_record *prev = NULL;
+	while (record != NULL) {
+		pldm_pdr_record *next = record->next;
+		struct pldm_pdr_hdr *hdr = (struct pldm_pdr_hdr *)record->data;
+		if ((record->is_remote == is_remote) &&
+		    hdr->type == PLDM_STATE_EFFECTER_PDR) {
+			struct pldm_state_effecter_pdr *pdr =
+				(struct pldm_state_effecter_pdr
+					 *)((uint8_t *)record->data);
+			if (pdr->effecter_id == effecter_id) {
+				delete_handle = hdr->record_handle;
+				if (repo->first == record) {
+					repo->first = next;
+				} else {
+					prev->next = next;
+				}
+				if (repo->last == record) {
+					repo->last = prev;
+					if (prev != NULL) {
+						prev->next = NULL;
+					}
+				}
+				--repo->record_count;
+				repo->size -= record->size;
+				if (record->data) {
+					free(record->data);
+				}
+				free(record);
+				break;
+			}
+			prev = record;
+
+		} else {
+			prev = record;
+		}
+		record = next;
+	}
+	return delete_handle;
+}
+
+LIBPLDM_ABI_TESTING
+uint16_t pldm_delete_by_sensor_id(pldm_pdr *repo, uint16_t sensor_id,
+				  bool is_remote)
+{
+	assert(repo != NULL);
+
+	uint32_t delete_handle = 0;
+	pldm_pdr_record *record = repo->first;
+	pldm_pdr_record *prev = NULL;
+	while (record != NULL) {
+		pldm_pdr_record *next = record->next;
+		struct pldm_pdr_hdr *hdr = (struct pldm_pdr_hdr *)record->data;
+		if ((record->is_remote == is_remote) &&
+		    hdr->type == PLDM_STATE_SENSOR_PDR) {
+			struct pldm_state_sensor_pdr *pdr =
+				(struct pldm_state_sensor_pdr
+					 *)((uint8_t *)record->data);
+			if (pdr->sensor_id == sensor_id) {
+				delete_handle = hdr->record_handle;
+				if (repo->first == record) {
+					repo->first = next;
+				} else {
+					prev->next = next;
+				}
+				if (repo->last == record) {
+					repo->last = prev;
+					if (prev != NULL) {
+						prev->next = NULL;
+					}
+				}
+				--repo->record_count;
+				repo->size -= record->size;
+				if (record->data) {
+					free(record->data);
+				}
+				free(record);
+				break;
+			}
+			prev = record;
+
+		} else {
+			prev = record;
+		}
+		record = next;
+	}
+	return delete_handle;
+}
+
+/* API to find the PDR record that is previous to a given PLDM PDR
+ * record in a given PLDM PDR repository
+ */
+static pldm_pdr_record *pldm_pdr_get_prev_record(pldm_pdr *repo,
+						 pldm_pdr_record *record)
+{
+	assert(repo);
+	assert(record);
+	pldm_pdr_record *prev = NULL;
+	pldm_pdr_record *curr = repo->first;
+	while (curr != NULL) {
+		if (curr->record_handle == record->record_handle) {
+			break;
+		}
+		prev = curr;
+		curr = curr->next;
+	}
+	return prev;
+}
+
+/* API to check if a PLDM PDR record is present in a PLDM PDR repository
+ */
+static bool is_record_present(pldm_pdr *repo, pldm_pdr_record *record)
+{
+	assert(repo);
+	assert(record);
+	if (pldm_pdr_get_prev_record(repo, record) || repo->first == record) {
+		return true;
+	}
+	return false;
+}
```

Comment:
This API is not doing as named. API tells `is_record_present()` but the API is checking the prev record or seeing if its a the first record and returning true/ false.

Reply:
will change to is_prev_record_present

Reply:
still not changed